vars/buildDockerAndPublishImage.groovy

```diff
@@ -3,25 +3,57 @@ import java.text.SimpleDateFormat
 import java.util.Date
 import java.text.DateFormat
 
+// makecall is a function to concentrate all the call to 'make'
+def makecall(String action, String imageDeployName, String targetOperationSystem, String specificDockerBakeFile) {
+  final String bakefileContent = libraryResource 'io/jenkins/infra/docker/jenkinsinfrabakefile.hcl'
+
+  // Please note that "make deploy" and the generated bake deploy file uses the environment variable "IMAGE_DEPLOY_NAME"
+  withEnv(["IMAGE_DEPLOY_NAME=${imageDeployName}", "TAG_NAME=${env.TAG_NAME}"]) {
+    if (isUnix()) {
+      if (! specificDockerBakeFile) {
+        specificDockerBakeFile = 'jenkinsinfrabakefile.hcl'
+        writeFile file: specificDockerBakeFile, text: bakefileContent
+      }
+      withEnv(["DOCKER_BAKE_FILE=${specificDockerBakeFile}"]) {
+        sh 'docker buildx create --use'
```

Comment:
=> Moving the `docker buildx create --use` instruction out from the make `build` target into the `makecall()` method had an unexpected side effect: the `make deploy-bake` rebuilds the whole images inside a new buildx builder (as seen in https://infra.ci.jenkins.io/job/docker-jobs/job/docker-confluence-data/job/PR-48/65/pipeline-console/?selected-node=123: there are NO `CACHED` docker build steps while https://infra.ci.jenkins.io/job/docker-jobs/job/docker-confluence-data/job/PR-48/63/pipeline-console/?selected-node=111 has them)

When it was part of the `make build`, it was called on once, while now it is called twice.


The suggestion belows solves this problem by using an idempotent shell command:

- By convention we want to use a buildx builder always named `buildx-builder`
- You can call this shell instruction once, twice or more, result will always be the same
- Hiding the `stderr` of the "use" instruction avoid worrying end users with an error message indication that it has to be created

```suggestion
        final String buildxBuilderName = buildx-builder
        sh "docker buildx use ${buildxBuilderName} 2>/dev/null || docker buildx create --use --name=${buildxBuilderName}'
```